### PR TITLE
[Container Registry] Transport wrapper

### PR DESCRIPTION
**Pull request by seankane-msft**
Addresses #17508 . Adding a transport wrapper to pipelines to ensure sessions are not closed when using context managers. PR is based off another branch not yet in master, need #17540  to be merged first.

### CODE REVIEW

File: sdk/containerregistry/azure-containerregistry/tests/test_container_registry_async.py [hunk 1]
```diff
@@ -62,3 +63,19 @@ async def test_delete_repository_does_not_exist(self, containerregistry_baseurl)
 
         with pytest.raises(ResourceNotFoundError):
             deleted_result = await client.delete_repository("not_real_repo")
+
+    @acr_preparer()
+    async def test_transport_closed_only_once(self, containerregistry_baseurl):
+        transport = AioHttpTransport()
+        client = self.create_registry_client(containerregistry_baseurl, transport=transport)
+        async with client:
```
Comment (iscai-msft) [inline sdk/containerregistry/azure-containerregistry/tests/test_container_registry_async.py:71]: I'm sorry idk how this test checks that the transport is only closed once :'(

If this isn't checking the transport's only closed once, you can look at @chlowell 's test here to gain inspo https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/keyvault/azure-keyvault-certificates/tests/test_context_manager.py#L10

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_base_client.py [hunk 2]
```diff
@@ -56,3 +57,28 @@ def close(self):
         Calling this method is unnecessary when using the client as a context manager.
         """
         self._client.close()
+
+
+class TransportWrapper(HttpTransport):
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_base_client.py:62]: Is there any async version of this?

File: sdk/containerregistry/azure-containerregistry/tests/test_container_registry_async.py (hunk 1 above)
Comment (kristapratico) [reply sdk/containerregistry/azure-containerregistry/tests/test_container_registry_async.py:71]: @seankane-msft can correct me, but I think we're testing that the child client does not close the transport of the parent client upon exiting the context manager

Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/tests/test_container_registry_async.py:71]: Krista is correct, I took inspo from storage https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/tests/test_common_blob.py#L2340-L2351

I can add an issue to address that situation in a later PR @iscai-msft

**Review by kristapratico — APPROVED** [review]

**Pull request merged**